### PR TITLE
For fenix#11679: submenu items order depends on appendExtensionSubMenuAtStart

### PR DESCRIPTION
**Pull request by psymoon**
For https://github.com/mozilla-mobile/fenix/issues/11679

---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt [hunk 1]
```diff
@@ -22,8 +22,8 @@ import mozilla.components.browser.state.store.BrowserStore
  * @param webExtIconTintColorResource Optional ID of color resource to tint the icons of back and
  * add-ons manager menu items.
  * @param onAddonsManagerTapped Callback to be invoked when add-ons manager menu item is selected.
- * @param appendExtensionActionAtStart true if web extensions appear at the top (start) of the menu,
- * false if web extensions appear at the bottom of the menu. Default to false (bottom).
+ * @param appendExtensionSubMenuAtStart true if web extension sub menu appear at the top (start) of
```
Comment (csadilek) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:25]: nit: I find this name and description confusing. :) Isn't it just switching the position of the `<- Add-ons` and `Add-onsManager` items? The extension actions (menu items) are always in-between those...

Comment (psymoon) [reply components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:25]: This is meant for the add-on sub menu position in the main menu. Sub menu items positions are not affected (hence the name change from action -> sub menu).

**Comment by psymoon** [comment]
bors r=csadilek

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt [hunk 2]
```diff
@@ -33,7 +33,7 @@ class WebExtensionBrowserMenuBuilder(
     private val store: BrowserStore,
     private val webExtIconTintColorResource: Int = NO_ID,
     private val onAddonsManagerTapped: () -> Unit = {},
-    private val appendExtensionActionAtStart: Boolean = false
+    private val appendExtensionSubMenuAtStart: Boolean = false
```
Comment (hakkikaancaliskan) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:36]: @psymoon This is breaking change on my side. Should we change that variable name in fenix?
https://github.com/mozilla-mobile/fenix/blob/3c2c7f705cb03895dcb090ae98fa9718cdeeef69/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt#L73

Comment (hakkikaancaliskan) [reply components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:36]: https://github.com/mozilla-mobile/fenix/pull/11979

**Pull request merged**